src/AssetsPage/CopyButton/index.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '@edx/paragon/src/Button';
+import copy from 'copy-to-clipboard';
+import styles from './CopyButton.scss';
+
+export default class CopyButton extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      wasClicked: false,
+    };
+
+    this.onClick = this.onClick.bind(this);
+    this.onBlur = this.onBlur.bind(this);
+    this.setButtonRef = this.setButtonRef.bind(this);
+  }
+
+  onClick() {
+    this.setState({
+      wasClicked: true,
+    });
+
+    copy(`${this.props.textToCopy}`);
+
+    /*
+      The above copy command removes focus from the button, so we
+      have to explicitly set it back. The copy command works by
+      inserting a span containing the text to be copied into the
+      DOM and running document.execCommand('copy') on it. Some
+      part of this process shifts focus.
+    */
+    this.buttonRef.focus();
+    this.props.onEvent(true);
```

Comment:
I'm finding `onEvent` to be a bit too general, since it's very much indicating the state of the button. I get that there can't be two `onClick` functions though. I honestly can't think of a better name though. Maybe `onCopyFocus` or `onCopyButtonClick`?

Reply:
Yeah, I find this part of the code kind of... un-indeal, in general. I think the difficulty in naming it reflects that. It's sort of a function that's responsible for both onClick and onBlur on the CopyButton. Is it okay if I go with `onCopyButtonClick` and then explain some of the ambiguity in a comment? I'm totally open to a different structure, but this was sort of necessitated by the need for a separate span on the page to handle screenreader text.

Reply:
Ok, sounds good! `onCopyButtonClick` makes sense (indicates that it's specific to this component) and a comment sounds good given the use case. Thank you!